Ability to run multiple LH tests and average results - similar to WPT #46

nenuadrian opened the pull request:
#46

Contain changes from https://github.com/gtech-professional-services/tagspeed_audit/pull/45 - which is to be merged first

<img width="1288" alt="image" src="https://user-images.githubusercontent.com/5632882/217029594-23d021be-478f-4942-a6d2-c0e7d9375ec9.png">


- [x] Tests pass
- [x] Appropriate changes to README are included in PR

adamread commented on src/core.ts:
```diff
@@ -147,16 +158,68 @@ async function extractRequestsFromPage(
+ * @param responses
+ */
+export function averageCrossReportMetrics(responses: LHResponse[]): LHResponse {
+  const FCP =
```
This could use an explanation in the comment. I've seen epsilon used for comparisons, but not like this.

nenuadrian replied:
I feel that wasn't needed there, tho it didn't hurt either. Epsilon is used generally for magin of error when comparing as accuracy in JS is questionable +/- epsilon https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/EPSILON#description

For averaging, I had seen in an example for averaging in JS that it was used but that was probably for further direct comparison in JS rather than for the human eye. It useful for programatic further comparisons to other number eg if you expect an average to be 5, you're better off checking for 5 +/- epsilon is my understanding.

adamread reviewed (approved):
LGTM other than the question about using epsilon in the average.

nenuadrian commented:
Fixed merge conflicts, a typo in HTML, removed epsilon and reset the result so far counter to 0 when an analysis is already on the screen as it was showing previous number until the baseline runs and overrides

adamread reviewed (dismissed):
LGTM

xpfgg commented on src/core.ts:
```diff
@@ -110,7 +110,9 @@ async function runLHForURL(
     }
 
-    fs.writeFileSync(`dist/${responses[0].reportUrl}`, lhr.report);
+    fs.writeFile(`dist/${responses[0].reportUrl}`, lhr.report, () => {
```
LGTM, maybe adding an error catch would be nice in case things don't go as smoothly as planned

nenuadrian replied:
There is one upstairs - but yes planning for a local one - but the observer API removes this report anyway - we'll see if we want to even keep it

xpfgg reviewed (approved):
LGTM

xpfgg reviewed (approved):
LGTM
